CDCgov/prime-reportstream
# 14601 authorization api

This PR adds authorization functionality to the auth and submission microservices.

Test Steps:
1. Get secrets and active access token from Jamie
2. Start up ReportStream, submissions, and auth (look at document in this PR for steps)
3. Submit a report to http://localhost:9000/api/v1/reports
4. Ensure the response you get is NOT 401 or 403. If something goes wrong in submissions thats beyond the scope of this ticket.

Bonus points: run the tests in the auth and submissions project to ensure they are passing on your machine locally.

## Changes
- Okta admin API calls to retrieve group information
- Added Okta-Groups header with a JWT value
- JWT read/writing
- Authorization logic for senders

## Checklist

### Testing

## Linked Issues
- Fixes #14601
